scripts/triggers/systems/dnd5e.mjs

```diff
@@ -0,0 +1,191 @@
+import { TRIGGERS } from '../../triggers.mjs'
+
+export default function init() {
+  DND5ETriggers.init();
+}
+
+export class DND5ETriggers {
+  /**
+   * Initialize the submodule.
+   */
+  static init() {
+    if (game.system.id !== "dnd5e") return;
+
+    // Inject our triggers into existing categories
+    const effectTriggers = TRIGGERS.find(t => t.label === "EFFECTMACRO.EffectTriggers");
+    const combatantTriggers = TRIGGERS.find(t => t.label === "EFFECTMACRO.CombatantTriggers");
+
+    effectTriggers.options.push(
+      "dnd5e.applyDamage",
+      "dnd5e.shortRest",
+      "dnd5e.longRest",
+      "dnd5e.beginConcentrating",
+      "dnd5e.endConcentration",
+      "dnd5e.rollHitDie"
+    );
+
+    combatantTriggers.options.push(
+      "dnd5e.rollAttack",
+      "dnd5e.rollDamage",
+      "dnd5e.rollInitiative"
+    );
+
+    // Add new section for roll-based triggers
+    TRIGGERS.push({
+      label: "EFFECTMACRO.RollTriggers",
+      options: [
+        "dnd5e.rollSavingThrow",
+        "dnd5e.rollDeathSave",
+        "dnd5e.rollAbilityCheck",
+        "dnd5e.rollSkill",
+        "dnd5e.rollToolCheck",
+        "dnd5e.rollConcentration"
+      ]
+    });
+
+    Hooks.on("dnd5e.rollAttack", DND5ETriggers._rollAttack);
+    Hooks.on("dnd5e.rollDamage", DND5ETriggers._rollDamage);
+    Hooks.on("dnd5e.rollSavingThrow", DND5ETriggers._rollSavingThrow);
+    Hooks.on("dnd5e.rollDeathSave", DND5ETriggers._rollDeathSave);
+    Hooks.on("dnd5e.rollAbilityCheck", DND5ETriggers._rollAbilityCheck);
+    Hooks.on("dnd5e.rollSkill", DND5ETriggers._rollSkill);
+    Hooks.on("dnd5e.rollToolCheck", DND5ETriggers._rollToolCheck);
+    Hooks.on("dnd5e.rollInitiative", DND5ETriggers._rollInitiative);
+    Hooks.on("dnd5e.rollConcentration", DND5ETriggers._rollConcentration);
+    Hooks.on("dnd5e.rollHitDie", DND5ETriggers._rollHitDie);
+    Hooks.on("dnd5e.restCompleted", DND5ETriggers._restCompleted);
+    Hooks.on("dnd5e.applyDamage", DND5ETriggers._applyDamage);
+    Hooks.on("dnd5e.beginConcentrating", DND5ETriggers._beginConcentrating);
+    Hooks.on("dnd5e.endConcentration", DND5ETriggers._endConcentration);
+  }
+
+  /**
+   * Utility method to filter and then call applicable effects with a trigger.
+   */
+  static async _filterAndCall(actor, hook, context) {
```

Comment:
The methods here are all underscored, not private. Are we expecting these to be subclassed?